src/v2/api/services/RoleService.ts

```diff
@@ -103,4 +105,46 @@ export class RoleService {
 
     await this.checkRole(userRoles, parent);
   }
-}
\ No newline at end of file
+
+
+  async transferRole (newStudentId: string, userId: string) {
+    const { id: groupId } = await this.groupRepository.find({
+      students: {
+        some: {
+          userId: {
+            in: [newStudentId, userId],
+          },
+        },
+      },
+    });
+
+    if (!groupId) {
+      throw new NoPermissionException();
+    }
+
+
+    const newStudentRole = await this.roleRepository.find({
+      groupRole: {
+        groupId: groupId,
+      },
+      userRoles: {
+        some: {
+          studentId: newStudentId,
+        },
+      },
+    });
+    const oldStudentRole = await this.roleRepository.find({
+      groupRole: {
+        groupId: groupId,
+      },
+      userRoles: {
+        some: {
+          studentId: userId,
+        },
+      },
+    });
+
+    await this.roleRepository.updateById(newStudentRole.id, { name: RoleName.CAPTAIN, weight: 100 });
+    await this.roleRepository.updateById(oldStudentRole.id, { name: RoleName.STUDENT, weight: 50 });
```

Comment:
Instead of updating the roles themselves, you should update the connection between the user and the role from the SAME group. In this way, you only need to change userRoles